### PR TITLE
Add mat4 vec4 mul and inv

### PR DESCRIPTION
**Pull request by TheFakeMontyOnTheRun**
Revert 3eb6f36ff7b170523fba2c26b438b5bf514c2a00, per @jroyalty request, so the original pull request from @Zubnix can be merged into master.

### CODE REVIEW

File: src/main/java/com/hackoeur/jglm/Mat4.java [hunk 1]
```diff
@@ -257,23 +257,23 @@ public int getNumRows() {
...
-			return (T) new Vec4(m00, m01, m02, m03);
-		case 1:
-			return (T) new Vec4(m10, m11, m12, m13);
-		case 2:
-			return (T) new Vec4(m20, m21, m22, m23);
-		case 3:
-			return (T) new Vec4(m30, m31, m32, m33);
-		default:
-			throw new IllegalArgumentException("Invalid column index = " + columnIndex);
+			case 0:
```
Comment (jroyalty) [inline src/main/java/com/hackoeur/jglm/Mat4.java]: Let's not change whitespace.

File: src/main/java/com/hackoeur/jglm/Matrices.java [hunk 2]
```diff
@@ -179,4 +179,116 @@ public static Mat4 rotate(final float phi, final Vec3 axis) {
...
+		final int n = 3;
+		float[][] inverted = invert(a,
+				n);
+		return new Mat3(inverted[0][0], inverted[1][0], inverted[2][0],
+				inverted[0][1], inverted[1][1], inverted[2][1],
+				inverted[0][2], inverted[1][2], inverted[2][2]);
+	}
+
+	private static float[][] invert(final float[][] a,
+									final int n){
```
Comment (jroyalty) [inline src/main/java/com/hackoeur/jglm/Matrices.java]: Let's collapse these lines.

**Comment by jroyalty** [comment]
So just a couple of things:

1. Sorry for the delay in reviewing this.
1. Thanks for removing the generic Mat param.
1. Why all the whitespace changes?  Can you adjust your editor and/or formatter to not make those.

I think if you can fixup those whitespace changes we are good.

Cheers!

File: src/main/java/com/hackoeur/jglm/Mat4.java (hunk 1 above)
Comment (TheFakeMontyOnTheRun) [inline src/main/java/com/hackoeur/jglm/Mat4.java]: Can you provide me with instructions on how the code should be formatted, please?

**Comment by TheFakeMontyOnTheRun** [comment]
Thanks for eventually answering.

Hey, I literally just took the code from the originals. Sure, I could use the default formatter for IntelliJ or Eclipse, but It would be more productive if I knew right away what's your taste for code style. Can you provide me a guideline?

File: src/main/java/com/hackoeur/jglm/Mat4.java (hunk 1 above)
Comment (jroyalty) [inline src/main/java/com/hackoeur/jglm/Mat4.java]: Well, in this particular instance, just as it was before this change.  In general, indent size = 4 using tabs.  If you are using Eclipse then I think the default formatter config should be fine.  As for the other parts of the diff involving whitespace, I think just leaving things formatted as they were before would be fine.

**Comment by jroyalty** [comment]
@TheFakeMontyOnTheRun My preference is really just that the codebase is consistently formatted.  Meaning, whitespace-only changes shouldn't be part of the diff.  I think if you just match the existing style it'll be fine.

**Comment by TheFakeMontyOnTheRun** [comment]
Oddly enough, IntelliJ  autoformatter changed all the lines in the files **except** the ones you noticed!
I will need more time to fix everything.

**Comment by TheFakeMontyOnTheRun** [comment]
Closing this as I created a new pull request that's a lot cleaner than this one.

**Pull request closed without merging**